WcaOnRails/app/models/user.rb

```diff
@@ -563,7 +565,7 @@ def admin?
   end
 
   def any_kind_of_delegate?
-    delegate_status.present?
+    delegate_status != nil && delegate_status != "trainee_delegate"
```

Comment:
I think we should leave this as is (see my comment on the issue [here](https://github.com/thewca/worldcubeassociation.org/issues/5118#issuecomment-597882239)), and perform the check only when doing the last final step of posting results/reports.